### ISSUE TITLE
Allow editing of run variables when relaunching a (failed) job

### ISSUE THREAD

**Issue by sagarkhushalani** [ansible/awx#14723]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Feature type

Enhancement to Existing Feature

### Feature Summary

We have the ability to relaunch jobs on either failed hosts or have it run on all hosts. However, when relaunching a job, we cannot edit the variables that were used to run the job. 

It may happen that a job failed because a variable was wrong, or that one of the variables controls output and we need it for better debugging the failure (both have happened for us).

Please add the ability to edit run-time variables when relaunching a job.

### Select the relevant components

- [X] UI
- [X] API
- [ ] Docs
- [ ] Collection
- [ ] CLI
- [ ] Other

### Steps to reproduce

1. Run a job
2. Go to job details page
3. Click "Relaunch"

### Current results

Variables JSON is shown, and is read-only. When clicking "Relaunch", there's no way to update values.

### Sugested feature result

Allow updating of variable values to a relaunch modal/page/dialog box.

**Comment by davidahern** [comment]
yes - please enable this to avoid having to re-enter everything when rerunning a job or bumping a version number

**Comment by watsonb** [comment]
+1 from me.  We have several templates with 10s of variables and, on occasion, someone mis-types a value.  Would be great to re-launch and get a "preview" screen where you had an opportunity to review and optionally change the value of a variable.

**Comment by fritz0011** [comment]
almost end of 2024 and still no progress on this.....

**Comment by HadadAvsha** [comment]
+1
